### PR TITLE
Feat/write acl api

### LINKED ISSUE #869: Integrate Write Access conversion
Add a method to the CoreDocument model (model/coredocument.go) to read values in CoreDocument and use them as input for the init rules/transitions...
This will break functional tests, which they will need to be updated as part of this work.

### PR DESCRIPTION
**Pull request by vedhavyas**
Closes #869

### CODE REVIEW

File: documents/coredocument.go [hunk 1]
```diff
@@ -486,70 +434,33 @@ func (cd *CoreDocument) coredocTree(docType string) (tree *proofs.DocumentTree,
...
 	wcs, err := cd.getWriteCollaborators(coredocumentpb.TransitionAction_TRANSITION_ACTION_EDIT)
 	if err != nil {
 		return CollaboratorsAccess{}, err
 	}
+
 	rc := filterCollaborators(rcs, filterIDs...)
 	wc := filterCollaborators(wcs, filterIDs...)
-
-	collabs := CollaboratorsAccess{
+	rc = filterCollaborators(rc, wc...)
```
Comment (vimukthi-git) [inline documents/coredocument.go]: switch the order of filtering rc and wc then you can avoid the 3rd filter here

File: documents/converters.go [hunk 2]
```diff
@@ -271,3 +272,87 @@ func FromP2PPaymentDetails(pdetails []*commonpb.PaymentDetails) ([]*PaymentDetai
...
+			if _, ok := wmap[c]; ok {
+				continue
+			}
+
+			wmap[c] = struct{}{}
+			wcs = append(wcs, c)
+		}
+	}
+
+	if racess != nil {
```
Comment (vimukthi-git) [inline documents/converters.go]: same logic as `waccess` so repeated code, Refactor to a function.

File: protobufs/invoice/service.proto [hunk 3]
```diff
@@ -60,30 +60,23 @@ message GetVersionRequest {
...
-  repeated string collaborators = 2;
-  InvoiceData data = 3;
+  document.ReadAccess read_access = 2;
+  document.WriteAccess write_access = 3;
+  InvoiceData data = 4;
 }
 
 message InvoiceResponse {
-  ResponseHeader header = 1;
+  document.ResponseHeader header = 1;
```
Comment (vimukthi-git) [inline protobufs/invoice/service.proto]: 👍

File: documents/converters.go (hunk 2 above)
Comment (vedhavyas) [reply documents/converters.go]: not the same logic ;)

**Pull request merged**